tests/api/runBlock.js

```diff
@@ -0,0 +1,123 @@
+const runTx = require('../../lib/runTx')
+const testData = require('./testdata.json')
+const { createGenesis, createAccount, setupVM } = require('./utils')
+const { setupPreConditions } = require('../util')
+
+function setup (vm = null) {
+  // Create a mock, if no real VM object provided.
+  if (vm === null) {
+    const stateManager = new StateManager()
+    vm = {
+      stateManager,
+      emit: (e, val, cb) => cb(),
+      populateCache: stateManager.warmCache.bind(stateManager),
+      runTx: (opts, cb) => cb(new Error('test')),
+      runCall: (opts, cb) => cb(new Error('test')),
+      _common: new Common('mainnet', 'byzantium')
+    }
+  }
+
+  return {
+    vm,
+    data: testData,
+    p: {
+      runBlock: promisify(runBlock.bind(vm)),
+      putAccount: promisify(vm.stateManager.putAccount.bind(vm.stateManager)),
+      cacheFlush: promisify(vm.stateManager.cache.flush.bind(vm.stateManager.cache))
+    }
+  }
+}
+
+tape('runBlock', async (t) => {
+  const suite = setup()
+
+  t.test('should fail without params', async (st) => {
+    suite.p.runBlock()
+      .then(() => st.fail('should have returned error'))
+      .catch((e) => st.ok(e.message.includes('invalid input'), 'correct error'))
+
+    st.end()
+  })
+
+  t.test('should fail without opts', async (st) => {
+    suite.p.runBlock({})
+      .then(() => st.fail('should have returned error'))
+      .catch((e) => st.ok(e.message.includes('invalid input'), 'correct error'))
+
+    st.end()
+  })
+
+  t.test('should fail when runTx fails', async (st) => {
+    const genesis = createGenesis()
+    const block = new Block(util.rlp.decode(suite.data.blocks[0].rlp))
+    await suite.p.runBlock({ block, root: genesis.header.stateRoot })
+      .then(() => t.fail('should have returned error'))
+      .catch((e) => t.equal(e.message, 'test'))
+
+    st.end()
+  })
+})
+
+tape('should fail when tx gas limit higher than block gas limit', async (t) => {
+  const suite = setup()
+
+  const genesis = createGenesis()
+  const block = new Block(util.rlp.decode(suite.data.blocks[0].rlp))
+  block.transactions[0].gasLimit = Buffer.from('3fefba', 'hex')
+  suite.p.runBlock({ block, root: genesis.header.stateRoot })
+    .then(() => t.fail('should have returned error'))
+    .catch((e) => t.ok(e.message.includes('higher gas limit')))
+
+  t.end()
+})
+
+tape('should fail when runCall fails', async (t) => {
+  const suite = setup()
+
+  const block = new Block(util.rlp.decode(suite.data.blocks[0].rlp))
+  // Add some balance to accounts, so they can run txes
+  for (let i = 0; i < block.transactions.length; i++) {
+    let tx = new Transaction(block.transactions[i])
+    const acc = createAccount()
+    await suite.p.putAccount(tx.from.toString('hex'), acc)
+  }
+  await suite.p.cacheFlush()
+
+  suite.vm.runTx = runTx
+  await suite.p.runBlock({ block, root: suite.vm.stateManager.trie.root })
+    .then(() => t.fail('should have returned error'))
+    .catch((e) => t.equal(e.message, 'test'))
```

Comment:
Again here, a bit more explicit stating why ``runCall`` fails so that others can easily grab would be nice.